### DISCUSSION TITLE
self-regulation and temp display on TRV, does VTherm adjust the offset/calibration variable?

### DISCUSSION THREAD

**Discussion by maia** [jmcollin78/versatile_thermostat#494] [Q&A]
It's been months since I last spent time with TRVs, but something has been nagging me the last winter: My Eve Thermo TRVs (Homekit) did not adjust the displayed target temperature based on the self-regulation. This led to lots of confusion as people turned down the TRV when they noticed that it was set to 25°C, even though that was just VTherm self-regulating.

I'm now considering the purchase of new TRVs with Zigbee protocol. Can VTherm now properly adjust the display on the device itself by changing the offset? For example the SONOFF TRVZB and also the AVATTO TR168 expose the variable `local_temperature_calibration` (see: https://www.zigbee2mqtt.io/devices/TRVZB.html and https://www.zigbee2mqtt.io/devices/ME168.html ) which defines the "Offset to add/subtract to the local temperature". Will VTherm use this variable in its self-regulation feature, which should allow keeping the target temperature displayed on the TRV synced with the target temperature in HA? Thanks!

**Comment by jmcollin78** [comment]
Hello, it is not planed today but this can be an interesting update. Do you have an updatable entity named "local_temperature_calibration" ?

**Reply by maia** [reply]
I haven't purchased these TRVs yet and if everything goes fine, my current TRVs (Eve Thermo) will have this entity available soon. @lboue is currently working on the implementation (thanks!), see https://community.home-assistant.io/t/eve-thermo-matter/693035/22

**Reply by Rhizix** [reply]
Would really like this as well, if you need any other info I'd gladly provide it (using SONOFF TRVZB)

**Reply by tbrasser** [reply]
Try the latest beta & set it up as over-climate :)

**Reply by Rhizix** [reply]
Ah, nice! I have two questions, though:

Why are the valve opening/closing angles required on the same setup screen where the Offset entity is defined?
I’d prefer not to use specific valve opening/closing degrees because, when my radiator valve is partially open (between 20-45%), there’s a noticeable noise caused by waterflow turbulence :/

The Offset entity seems to work, but I suspect something else is not functioning correctly in this beta build. The Vtherm entity consistently shows the "heating" state, even though the target temperature is well below the current temperature.


![image](https://github.com/user-attachments/assets/e7cb9438-1e94-4903-a434-037d80d5fe27)

**Reply by tbrasser** [reply]
what's the valve status? If it's above 0% it will show as heating.

**Reply by Rhizix** [reply]
It was indeed at 15%, but why is it open when the temperature is above the setpoint?
I’d also like the option to skip defining the valve entities and simply use the basic on/off climate mode.
It feels like these two features -the offset and the valve opening/closing method- shouldn’t be really linked.

**Reply by tbrasser** [reply]
They're not, you can probably leave the valve entities unfilled.

**Reply by tbrasser** [reply]
If you have auto regulation on it may open the valve to prevent undershoot. Check in on the trvzb issue/topic (Y)

**Reply by Rhizix** [reply]
The Valve Opening one seems to be mandatory.
![image](https://github.com/user-attachments/assets/5dab814f-e32e-486a-9e18-c9ac5773ce2b)

It’s likely because that options menu only becomes available when "Auto-regulation" is set to "Direct control of Valve." However, as mentioned earlier, that’s not something I actually want to use.
